src/ray/common/task/task_spec.h

```diff
@@ -17,6 +17,9 @@ extern "C" {
 
 namespace ray {
 
+typedef std::vector<std::string> FunctionDescriptor;
+typedef std::pair<ResourceSet, FunctionDescriptor> SchedulingClass;
```

Comment:
We could also make the Scheduling class a class or struct, that might clean up the code a little and avoids pairs with too many elements.

Reply:
Call depth isn't part of the task spec right? It might be a bit involved to track that.

I think function name vs call depth is somewhat orthogonal, ideally you would have both. It is possible to have starvation issues in either case.

Also re: struct, I prefer std::pair when possible since it's the minimal structure required.

Reply:
Yeah, it's not part of the task spec, although it wouldn't be that difficult to track.

We can leave it for a later PR, but I'd leave the nested tasks issue open if we're not going to do it now.

Reply:
Hm you mean if you recursively call the same task? I agree that is still an open problem, but is different from the one reported by the users.